File: bkmonitor/packages/monitor_web/commons/robot/constant.py
# -*- coding: utf-8 -*-
"""
Tencent is pleased to support the open source community by making 蓝鲸智云 - 监控平台 (BlueKing - Monitor) available.
Copyright (C) 2017-2021 THL A29 Limited, a Tencent company. All rights reserved.
Licensed under the MIT License (the "License"); you may not use this file except in compliance with the License.
You may obtain a copy of the License at http://opensource.org/licenses/MIT
Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
specific language governing permissions and limitations under the License.
"""

HOUR = 60 * 60

ROBOT_AI_SETTING_KEY = "robot"
LATEST_FETCH_TIME_KEY = "latest_fetch_time"

MIN_FETCH_TIME_RANGE = HOUR  # 最小拉取间隔一小时
MAX_FETCH_TIME_RANGE = HOUR * 24  # 最大拉取间隔一天


class RobotLevel:
    RED = 1
    YELLOW = 2
    BLUE = 3


DISPLAY_TIME_LEVEL = [
    HOUR * 1,
    HOUR * 3,
    HOUR * 6,
    HOUR * 12,
    HOUR * 24,
]

File: bkmonitor/webpack/src/monitor-pc/store/modules/constant.js
/*
 * Tencent is pleased to support the open source community by making
 * 蓝鲸智云PaaS平台 (BlueKing PaaS) available.
 *
 * Copyright (C) 2021 THL A29 Limited, a Tencent company.  All rights reserved.
 *
 * 蓝鲸智云PaaS平台 (BlueKing PaaS) is licensed under the MIT License.
 *
 * License for 蓝鲸智云PaaS平台 (BlueKing PaaS):
 *
 * ---------------------------------------------------
 * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
 * documentation files (the "Software"), to deal in the Software without restriction, including without limitation
 * the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
 * to permit persons to whom the Software is furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all copies or substantial portions of
 * the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
 * THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF
 * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
 * IN THE SOFTWARE.
 */
const state = {
  eventStatusMap: {
    30: 'ABNORMAL', // 异常事件
    20: 'RECOVERED', // 已恢复
    10: 'CLOSED', // 已恢复
  },
  eventLevelMap: {
    1: '致命',
    2: '预警',
    3: '提醒',
  },
};

const getters = {
  eventStatusMap(state) {
    return state.eventStatusMap;
  },
  eventLevelMap(state) {
    return state.eventLevelMap;
  },
};

export default {
  namespaced: true,
  state,
  getters,
};
